pronto_mia/lib/ui/views/deployment_plan/overview/deployment_plan_overview_view.dart

```diff
@@ -15,100 +18,97 @@ class DeploymentPlanOverviewView extends StatelessWidget {
-            child: ListView.builder(
-              itemCount: model.data.length,
-              itemBuilder: (context, index) {
-                final dateFormat = DateFormat('dd.MM.yyyy');
-                final dateTimeFormat = DateFormat('dd.MM.yyyy hh:mm');
-                final availableFromDateFormatted =
-                    dateFormat.format(model.data[index].availableFrom);
-                final availableFromFormatted =
-                    dateTimeFormat.format(model.data[index].availableFrom);
-                final availableUntilFormatted =
-                    dateTimeFormat.format(model.data[index].availableUntil);
+  Widget _buildListItem(
+    BuildContext context,
+    DeploymentPlanOverviewViewModel model,
+    DeploymentPlan deploymentPlan,
+  ) {
+    final dateFormat = DateFormat('dd.MM.yyyy');
+    final dateTimeFormat = DateFormat('dd.MM.yyyy hh:mm');
+    final availableFromDateFormatted =
+        dateFormat.format(deploymentPlan.availableFrom);
+    final availableFromFormatted =
+        dateTimeFormat.format(deploymentPlan.availableFrom);
+    final availableUntilFormatted =
+        dateTimeFormat.format(deploymentPlan.availableUntil);
 
-                return Card(
-                  child: ListTile(
-                    title: Text(model.data[index].description ??
-                        'Einsatzplan $availableFromDateFormatted'),
-                    subtitle: Text(
-                      '$availableFromFormatted - $availableUntilFormatted',
-                    ),
-                    onTap: () {
-                      if (adminModeEnabled) {
-                        model.editDeploymentPlan(model.data[index]);
-                      } else {
-                        model.openPdf(model.data[index]);
-                      }
-                    },
-                  ),
-                );
-              },
-            ),
-          );
-        })(),
-        floatingActionButton: adminModeEnabled
-            ? FloatingActionButton(
-                tooltip: 'Einsatzplan erstellen',
-                backgroundColor: Colors.green,
-                onPressed: model.createDeploymentPlan,
-                heroTag: null,
-                child: const Icon(Icons.post_add),
-              )
-            : null,
-        floatingActionButtonLocation: (() {
-          if (kIsWeb) {
-            return FloatingActionButtonLocation.endFloat;
+    return Card(
+      child: ListTile(
+        title: Text(deploymentPlan.description ??
+            'Einsatzplan $availableFromDateFormatted'),
+        subtitle: Text(
+          '$availableFromFormatted - $availableUntilFormatted',
+        ),
+        onTap: () {
+          if (adminModeEnabled) {
+            model.editDeploymentPlan(
+              deploymentPlan: deploymentPlan,
+              asDialog: getValueForScreenType<bool>(
+                context: context,
+                mobile: false,
+                desktop: true,

```

Comment:
Why is this hardcoded? Do you know your context at this point?

Reply:
`getValueForScreenType` decices the current context for me and returns the correspondig boolean value i have defined for each screentype.